Sage/carbon
# fix(select): reset height and width after closing

The useFloating hook, which is used by Select to position the list, correctly resets various CSS properties to their previous values when the component is no longer open. But it failed to do this with width or height. Since the select list computes its width when opened, this lack of cleanup results in a hard-coded CSS width in the style attribute, which messes up the layout if any later resizing happens. This is now fixed by cleaning up the width and height properties on close.

fix #5981

### Proposed behaviour



After opening and closing a select, there should be no `style` attribute with a `width` or `height` property (if there was not one before).

### Current behaviour

A `width` style is added to the select textbox element when opened, with a hardcoded pixel value - and this is not removed when the select is closed. This results in any later resizes - whether from screenresize or later adjustments to flex containers - not behaving correctly, and sudden "jumps" to the correct position if the select is later reopened.



### Checklist




#### QA


### Additional context



### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/strange-microservice-6sxvi2?file=/src/App.tsx:227-723



To observe the issue(s) on master with the above codesandbox:
- open the result panel in a new tab (this makes it easier to observe what is going on)
- first trying resizing the page (make it smaller, then bigger). Both columns (the one with the selects and the one with the plain text) should stay at 50% of the screen width, as intended.
- now try opening and closing one of the selects, and doing the same. You should see the section with the selects maintain a constant width, with the other one shrinking, and eventually disappearing off-screen when the screen is small enough.
- now try opening the same select as previously opened. You should see the layout suddenly snap back to what it should be - both sections taking 50% of the width.
- finally, having done this, widen the window and see that the just-opened select maintains its width and therefore becomes shorter than the other inputs. Again this is "fixed" by re-opening the select.

The issue can also be observed on Storybook with eg https://carbon.sage.com/?path=/docs/select--default (or any of the stories with an inline label). First open and close the select, then shrink the screen width - you should see the select gradually take up more and more of the available space (because its width is staying fixed), when it should shrink in proportion. Reopening the select at any point makes it jump back to the desired width.

All of these issues should be fixed by the code-changes in this PR - please check for this, as well as regression testing on Select and the other components which use the `useFloating` hook: Tooltip, ActionPopover, Date, MultiActionButton, PopoverContainer, SplitButton.
